src/script/script.cpp

```diff
@@ -227,3 +238,29 @@ bool CScript::IsPushOnly() const
     }
     return true;
 }
+
+int CScript::Type() const
+{
+    if (this->IsPayToPublicKeyHash())
+        return 1;
+    if (this->IsPayToScriptHash())
+        return 2;
+    // We don't know this script
+    return 0;
+}
+
+// This method should only be called if Type() returned nonzero
+vector<unsigned char> CScript::AddressHash() const
+{
+    int start = 0;
+    if (this->IsPayToScriptHash()) {
+        start = 2;
+    } else if (this->IsPayToPublicKeyHash()) {
+        start = 3;
+    } else {
+        assert(false);
```

Comment:
So it looks like a nonstandard output script could trigger the assertion and termiante the app.  So maybe this method only gets called once we're already sure the script is valid, per the comment.  If yes, then if you already have the type (1 or 2) from calling Type(), why not pass that value in and check against it, instead of calling IsPayToXXXHash() again?

Reply:
I agree with the removal of the assertion, but I prefer to just call `this->IsPayToPublicKeyHash()` again rather than to pass in the type. See my other comment related to this.